### PR TITLE
zh,en: add compact and multiple-rows for v5.3.0

### PR DESCRIPTION
**Pull request by GMHDBJD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-dm) that's required for repo owners to accept my contribution.

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: en/glossary.md [hunk 1]
```diff
@@ -93,14 +93,14 @@ In the case of clearly mentioning "incremental", use replicate/replication inste
 
 ### Safe mode
 
-Safe mode is the mode in which DML statements can be imported more than once when the primary key or unique index exists in the table schema. In this mode, some statements from the upstream are migrated to the downstream only after they are re-written. The `INSERT` statement is re-written as `REPLACE`; the `UPDATE` statement is re-written as `DELETE` and `REPLACE`.
+Safe mode is the mode in which DML statements can be imported more than once when the primary key or unique index exists in the table schema. In this mode, some statements from the upstream are migrated to the downstream only after they are re-written. Before v5.3.0, the `INSERT` statement is re-written as `REPLACE`; the `UPDATE` statement is re-written as `DELETE` and `REPLACE`. After v5.3.0, the `INSERT` statement is re-written as `INSERT ON DUPLICATE KEY UPDATE`; the `UPDATE` statement is re-written as `DELETE` and `INSERT ON DUPLICATE KEY UPDATE`.
 
 This mode is enabled in any of the following situations:
 
-- TiDB DM automatically enables the safe mode within 1 minutes immediately after the incremental replication task is started or resumed.
```
Comment (glorv) [inline en/glossary.md:100]: We should reserve this description(and change to a properer way)  as it's a import aspect to let our users to know how we ensure compatibility and correctness when task resume.

Comment (glorv) [reply en/glossary.md:100]: /cc @lichunzhu PTAL

File: zh/task-configuration-file-full.md [hunk 2]
```diff
@@ -110,7 +110,19 @@ syncers:                             # sync 处理单元的运行配置参数
     worker-count: 16                 # 应用已传输到本地的 binlog 的并发线程数量，默认值为 16。调整此参数不会影响上游拉取日志的并发，但会对下游产生显著压力。
     batch: 100                       # sync 迁移到下游数据库的一个事务批次 SQL 语句数，默认值为 100，建议一般不超过 500。
     enable-ansi-quotes: true         # 若 `session` 中设置 `sql-mode: "ANSI_QUOTES"`，则需开启此项
-    safe-mode: false                 # 设置为 true，则将来自上游的 `INSERT` 改写为 `REPLACE`，将 `UPDATE` 改写为 `DELETE` 与 `REPLACE`，保证在表结构中存在主键或唯一索引的条件下迁移数据时可以重复导入 DML。在启动或恢复增量复制任务的前 1 分钟（在 v2.0.3 及之前的版本中为 5 分钟）内 TiDB DM 会自动启动 safe mode
+
+    # 设置为 true，v5.3.0 之前，将来自上游的 `INSERT` 改写为 `REPLACE`，将 `UPDATE` 改写为 `DELETE` 与 `REPLACE`。将 `INSERT` 改写为 `INSERT ON DUPLICATE KEY UPDATE`，将 `UPDATE` 改写为 `DELETE` 与 `INSERT ON DUPLICATE KEY UPDATE`。保证在表结构中存在主键或唯一索引的条件下迁移数据时可以重复导入 DML。
+    safe-mode: false
+    # 设置为 true，DM 会在不增加延迟的情况下，尽可能的将上游对同一条数据的多次操作压缩成一次操作。
```
Comment (Liuxiaozhen12) [inline zh/task-configuration-file-full.md:116]:
```suggestion
    # 设置为 true，DM 会在不增加延迟的情况下，尽可能地将上游对同一条数据的多次操作压缩成一次操作。
```

File: zh/task-configuration-file-full.md [hunk 3]
```diff
@@ -110,7 +110,19 @@ syncers:                             # sync 处理单元的运行配置参数
...
+    # 如 INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1; 会被压缩成 INSERT INTO tb(a,b) VALUES(1,11); 其中 a 为主键
+    # 如 UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1; 会被压缩成 UPDATE tb(a,b) SET b=2 WHERE a=1; 其中 a 为主键
+    # 如 DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1); 会被压缩成 INSERT INTO tb(a,b) VALUES(1,1); 其中 a 为主键
+    compact: false
+    # 设置为 true，DM 会尽可能的将多条同类型的语句合并到一条语句中，生成一条带多行数据的 SQL 语句。
```
Comment (Liuxiaozhen12) [inline zh/task-configuration-file-full.md:121]:
```suggestion
    # 设置为 true，DM 会尽可能地将多条同类型的语句合并到一条语句中，生成一条带多行数据的 SQL 语句。
```

File: zh/task-configuration-file-full.md [hunk 4]
```diff
@@ -110,7 +110,19 @@ syncers:                             # sync 处理单元的运行配置参数
...
+    # 如 INSERT INTO tb(a,b) VALUES(1,1); INSERT INTO tb(a,b) VALUES(2,2); 会变成 INSERT INTO tb(a,b) VALUES(1,1),(2,2);
+    # 如 UPDATE tb SET b=11 WHERE a=1; UPDATE tb(a,b) set b=22 WHERE a=2; 会变成 INSERT INTO tb(a,b) VALUES(1,11),(2,22) ON DUPLICATE KEY UPDATE a=VALUES(a), b=VALUES(b); 其中 a 为主键
+    # 如 DELETE FROM tb WHERE a=1; DELETE FROM tb WHERE a=2 会变成 DELETE FROM tb WHERE (a) IN (1),(2)。其中 a 为主键
```
Comment (Liuxiaozhen12) [inline zh/task-configuration-file-full.md:124]:
```suggestion
    # 如 DELETE FROM tb WHERE a=1; DELETE FROM tb WHERE a=2 会变成 DELETE FROM tb WHERE (a) IN (1),(2)；其中 a 为主键
```

File: en/glossary.md [hunk 5]
```diff
@@ -93,14 +93,14 @@ In the case of clearly mentioning "incremental", use replicate/replication inste
...
 - The safe mode remains enabled when the `safe-mode` parameter in the task configuration file is set to `true`.
 - In shard merge scenarios, the safe mode remains enabled before DDL statements are replicated in all sharded tables.
 - If the argument `--consistency none` is configured for the dump processing unit of a full migration task, it cannot be determined whether the binlog changes at the beginning of the export affect the exported data or not. Therefore, the safe mode remains enabled for the incremental replication of these binlog changes.
+- The task paused by error and some data may be executed twice after resumed.
```
Comment (Liuxiaozhen12) [inline en/glossary.md:103]:
```suggestion
- If the task is paused by error and then resumed, some data may be executed twice.
```

File: en/task-configuration-file-full.md [hunk 6]
```diff
@@ -115,7 +115,19 @@ syncers:
...
+
+    # If set to true, before v5.3.0, `INSERT` statements from upstream are rewritten to `REPLACE` statements, and `UPDATE` statements are rewritten to `DELETE` and `REPLACE` statements. After v5.3.0, `INSERT` statements are rewritten to `INSERT ON DUPLICATE KEY UPDATE` statements, and `UPDATE` statements are rewritten to `DELETE` and `INSERT ON DUPLICATE KEY UPDATE` statements. This ensures that DML statements can be imported repeatedly during data migration when there is any primary key or unique index in the table schema.
+    safe-mode: false
+    # If set to true, DM will compact as many upstream statements on the same rows as possible into a single statements without increasing latency.
+    # such as `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, a is the primary key
+    # `UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1;` will be compacted to `UPDATE tb(a,b) SET b=2 WHERE a=1;`, a is the primary key
+    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `INSERT INTO tb(a,b) VALUES(1,1);`, a is the primary key
```
Comment (Liuxiaozhen12) [inline en/task-configuration-file-full.md:124]:
```suggestion
    # If set to true, DM compacts as many upstream statements on the same rows as possible into a single statements without increasing latency.
    # For example, `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, and "a" is the primary key
    # `UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1;` will be compacted to `UPDATE tb(a,b) SET b=2 WHERE a=1;`, and "a" is the primary key
    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `INSERT INTO tb(a,b) VALUES(1,1);`, and "a" is the primary key
```

File: en/task-configuration-file-full.md [hunk 7]
```diff
@@ -115,7 +115,19 @@ syncers:
...
+    compact: false
+    # if set to true, DM will combine as many statements of the same type as possible into a single statement and generate a single SQL statement with multiple rows of data.
+    # such as `INSERT INTO tb(a,b) VALUES(1,1); INSERT INTO tb(a,b) VALUES(2,2);` will become `INSERT INTO tb(a,b) VALUES(1,1),(2,2);`
+    # `UPDATE tb SET b=11 WHERE a=1; UPDATE tb(a,b) set b=22 WHERE a=2;` will become `INSERT INTO tb(a,b) VALUES(1,11),(2,22) ON DUPLICATE KEY UPDATE a=VALUES(a), b= VALUES(b);`, a is the primary key
+    # `DELETE FROM tb WHERE a=1; DELETE FROM tb WHERE a=2` will become `DELETE FROM tb WHERE (a) IN (1),(2)`, a is the primary key
```
Comment (Liuxiaozhen12) [inline en/task-configuration-file-full.md:129]:
```suggestion
    # If set to true, DM combines as many statements of the same type as possible into a single statement and generates a single SQL statement with multiple rows of data.
    # For example, `INSERT INTO tb(a,b) VALUES(1,1); INSERT INTO tb(a,b) VALUES(2,2);` will become `INSERT INTO tb(a,b) VALUES(1,1),(2,2);`
    # `UPDATE tb SET b=11 WHERE a=1; UPDATE tb(a,b) set b=22 WHERE a=2;` will become `INSERT INTO tb(a,b) VALUES(1,11),(2,22) ON DUPLICATE KEY UPDATE a=VALUES(a), b= VALUES(b);`, and "a" is the primary key
    # `DELETE FROM tb WHERE a=1; DELETE FROM tb WHERE a=2` will become `DELETE FROM tb WHERE (a) IN (1),(2)`, and "a" is the primary key
```

**Review by Liuxiaozhen12 — APPROVED** [review]

File: en/glossary.md (hunk 1 above)
Comment (lichunzhu) [reply en/glossary.md:100]: How about this description? But this seems too long.
```
- TiDB DM automatically enables the safe mode within 1 minutes immediately after the incremental replication task is resumed from a downstream database failture.
```

Comment (GMHDBJD) [reply en/glossary.md:100]: Does https://github.com/pingcap/docs-dm/blame/54340248eac298c44e1e9df396933ed9469515b9/en/glossary.md#L103 clearly describe?

Comment (lichunzhu) [reply en/glossary.md:100]: I think that's enough.

**Review by glorv — APPROVED** [review]

File: en/task-configuration-file-full.md [hunk 8]
```diff
@@ -115,7 +115,19 @@ syncers:
...
+
+    # If set to true, `INSERT` statements from upstream are rewritten to `REPLACE` statements, and `UPDATE` statements are rewritten to `DELETE` and `REPLACE` statements. This ensures that DML statements can be imported repeatedly during data migration when there is any primary key or unique index in the table schema.
+    safe-mode: false
+    # If set to true, DM compacts as many upstream statements on the same rows as possible into a single statements without increasing latency.
+    # For example, `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, and "a" is the primary key
+    # `UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1;` will be compacted to `UPDATE tb(a,b) SET b=2 WHERE a=1;`, and "a" is the primary key
+    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `INSERT INTO tb(a,b) VALUES(1,1);`, and "a" is the primary key
```
Comment (lance6716) [inline en/task-configuration-file-full.md:124]:
```suggestion
    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `REPLACE INTO tb(a,b) VALUES(1,1);`, and "a" is the primary key
```

File: en/task-configuration-file-full.md [hunk 9]
```diff
@@ -115,7 +115,19 @@ syncers:
...
-    safe-mode: false                 # If set to true, `INSERT` statements from upstream are rewritten to `REPLACE` statements, and `UPDATE` statements are rewritten to `DELETE` and `REPLACE` statements. This ensures that DML statements can be imported repeatedly during data migration when there is any primary key or unique index in the table schema. TiDB DM automatically enables safe mode within the first minute (in v2.0.3 and earlier versions, it is 5 minutes) after starting or resuming migration tasks.
+
+    # If set to true, `INSERT` statements from upstream are rewritten to `REPLACE` statements, and `UPDATE` statements are rewritten to `DELETE` and `REPLACE` statements. This ensures that DML statements can be imported repeatedly during data migration when there is any primary key or unique index in the table schema.
+    safe-mode: false
+    # If set to true, DM compacts as many upstream statements on the same rows as possible into a single statements without increasing latency.
+    # For example, `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, and "a" is the primary key
```
Comment (lance6716) [inline en/task-configuration-file-full.md:122]:
```suggestion
    # For example, `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, where "a" is the primary key
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: en/task-configuration-file-full.md [hunk 10]
```diff
@@ -115,7 +115,19 @@ syncers:
...
+
+    # If set to true, `INSERT` statements from upstream are rewritten to `REPLACE` statements, and `UPDATE` statements are rewritten to `DELETE` and `REPLACE` statements. This ensures that DML statements can be imported repeatedly during data migration when there is any primary key or unique index in the table schema.
+    safe-mode: false
+    # If set to true, DM compacts as many upstream statements on the same rows as possible into a single statements without increasing latency.
+    # For example, `INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1`;` will be compacted to `INSERT INTO tb(a,b) VALUES(1,11);`, where "a" is the primary key
+    # `UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1;` will be compacted to `UPDATE tb(a,b) SET b=2 WHERE a=1;`, and "a" is the primary key
```
Comment (lance6716) [inline en/task-configuration-file-full.md:123]:
```suggestion
    # `UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1;` will be compacted to `UPDATE tb(a,b) SET b=2 WHERE a=1;`, where "a" is the primary key
```

File: en/task-configuration-file-full.md [hunk 11]
```diff
@@ -115,7 +115,19 @@ syncers:
...
+    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `REPLACE INTO tb(a,b) VALUES(1,1);`, and "a" is the primary key
```
Comment (lance6716) [inline en/task-configuration-file-full.md:124]:
```suggestion
    # `DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1);` will be compacted to `REPLACE INTO tb(a,b) VALUES(1,1);`, where "a" is the primary key
```

File: zh/task-configuration-file-full.md [hunk 12]
```diff
@@ -110,7 +110,19 @@ syncers:                             # sync 处理单元的运行配置参数
...
     batch: 100                       # sync 迁移到下游数据库的一个事务批次 SQL 语句数，默认值为 100，建议一般不超过 500。
     enable-ansi-quotes: true         # 若 `session` 中设置 `sql-mode: "ANSI_QUOTES"`，则需开启此项
-    safe-mode: false                 # 设置为 true，则将来自上游的 `INSERT` 改写为 `REPLACE`，将 `UPDATE` 改写为 `DELETE` 与 `REPLACE`，保证在表结构中存在主键或唯一索引的条件下迁移数据时可以重复导入 DML。在启动或恢复增量复制任务的前 1 分钟（在 v2.0.3 及之前的版本中为 5 分钟）内 TiDB DM 会自动启动 safe mode
+
+    # 设置为 true，则将来自上游的 `INSERT` 改写为 `REPLACE`，将 `UPDATE` 改写为 `DELETE` 与 `REPLACE`，保证在表结构中存在主键或唯一索引的条件下迁移数据时可以重复导入 DML。
+    safe-mode: false
+    # 设置为 true，DM 会在不增加延迟的情况下，尽可能地将上游对同一条数据的多次操作压缩成一次操作。
+    # 如 INSERT INTO tb(a,b) VALUES(1,1); UPDATE tb SET b=11 WHERE a=1; 会被压缩成 INSERT INTO tb(a,b) VALUES(1,11); 其中 a 为主键
+    # 如 UPDATE tb SET b=1 WHERE a=1; UPDATE tb(a,b) SET b=2 WHERE a=1; 会被压缩成 UPDATE tb(a,b) SET b=2 WHERE a=1; 其中 a 为主键
+    # 如 DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1); 会被压缩成 INSERT INTO tb(a,b) VALUES(1,1); 其中 a 为主键
```
Comment (lance6716) [inline zh/task-configuration-file-full.md:119]:
```suggestion
    # 如 DELETE FROM tb WHERE a=1; INSERT INTO tb(a,b) VALUES(1,1); 会被压缩成 REPLACE INTO tb(a,b) VALUES(1,1); 其中 a 为主键
```

**Review by lance6716 — APPROVED** [review]

File: zh/glossary.md [hunk 13]
```diff
@@ -99,10 +99,10 @@ DM-worker 内部用于从上游拉取 Binlog 并写入数据到 Relay log 的处
 
 该模式会在满足如下任一条件时启用：
 
-- 启动或恢复增量复制任务的前 1 分钟保持启用
 - 任务配置文件中设置 `safe-mode: true` 时会始终启用
 - 合库合表模式下，DDL 尚未在所有分表完成同步时保持启用
 - 在全量迁移任务中的 dump 处理单元配置 `--consistency none` 后，不能确定导出开始时的 binlog 变动是否影响了导出数据。Safe mode 会在增量复制这部分 binlog 时保持启用
+- 任务出错停止并恢复后，有些数据可能会被执行两次时保持启用
```
Comment (Liuxiaozhen12) [inline zh/glossary.md:105]:
```suggestion
- 任务出错停止并恢复后，对有些数据的操作可能会被执行两次时保持启用
```

File: en/glossary.md [hunk 14]
```diff
@@ -97,10 +97,10 @@ Safe mode is the mode in which DML statements can be imported more than once whe
 
 This mode is enabled in any of the following situations:
 
-- TiDB DM automatically enables the safe mode within 1 minutes immediately after the incremental replication task is started or resumed.
 - The safe mode remains enabled when the `safe-mode` parameter in the task configuration file is set to `true`.
 - In shard merge scenarios, the safe mode remains enabled before DDL statements are replicated in all sharded tables.
 - If the argument `--consistency none` is configured for the dump processing unit of a full migration task, it cannot be determined whether the binlog changes at the beginning of the export affect the exported data or not. Therefore, the safe mode remains enabled for the incremental replication of these binlog changes.
+- If the task is paused by error and then resumed, some data may be executed twice.
```
Comment (Liuxiaozhen12) [inline en/glossary.md:103]:
```suggestion
- If the task is paused by error and then resumed, the operations on some data might be executed twice.
```

**Review by Liuxiaozhen12 — APPROVED** [review]
rest LGTM

**Pull request merged**